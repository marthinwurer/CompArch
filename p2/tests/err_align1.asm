;
; tests unaligned memory data references
;
; uses instructions MOV, HALT
;
; uses addressing modes 0, 6
;
	.globl	start
v1:	.word	10
v2:	.byte	5
v3:	.byte	10
. = . + 50
;
start:	mov	v1,r1
	mov	v3,r2
	halt
;
	.end	start
