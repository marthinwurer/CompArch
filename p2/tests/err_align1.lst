      1                                 ;
      2                                 ; tests unaligned memory data references
      3                                 ;
      4                                 ; uses instructions MOV, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 6
      7                                 ;
      8 000000                                   .globl  start
      9 000000  000010                  v1:      .word   10
     10 000002     005                  v2:      .byte   5
     11 000003     010                  v3:      .byte   10
     12 000054                          . = . + 50
     13                                 ;
     14 000054  016701  177720          start:   mov     v1,r1
     15 000060  016702  177717                   mov     v3,r2
     16 000064  000000                           halt
     17                                 ;
     18 000066  000054'                          .end    start


pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000054 RG 
v1       000000 R  
v2       000002 R  
v3       000003 R  
