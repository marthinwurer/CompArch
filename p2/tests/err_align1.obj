000 2 08 00
002 2 05 00
003 2 08 00
02c 4 c1 1d d0 ff
030 4 c2 1d cf ff
034 2 00 00
02c
