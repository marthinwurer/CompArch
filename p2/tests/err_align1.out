CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 54
000054:  PS[00]  016701 177720  MOV 177720(R7),R1
000060:  PS[00]  016702 177717  MOV 177717(R7),R2

Machine Halted - unaligned memory access

Simulated time 21 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
